src/core/mount.c

```diff
@@ -517,6 +493,13 @@ static int mount_add_default_dependencies(Mount *m) {
         if (r < 0)
                 return r;
 
+        /* If this is a tmpfs mount then we have to unmount it before we try to deactivate swaps */
+        if (streq(p->fstype, "tmpfs")) {
```

Comment:
`p->fstype` may be NULL. I am not sure but I guess that `streq_ptr()` is suitable here.

Reply:
By changing to `streq_ptr()`, my system boots and works fine.